src/lightning_app/utilities/packaging/cloud_compute.py

```diff
@@ -104,8 +108,15 @@ def to_dict(self):
         return {"type": __CLOUD_COMPUTE_IDENTIFIER__, **asdict(self)}
 
     @classmethod
-    def from_dict(cls, d):
+    def from_dict(cls, d: dict):
         assert d.pop("type") == __CLOUD_COMPUTE_IDENTIFIER__
+        mounts = d.pop("mount", None)
+        if isinstance(mounts, dict):
```

Comment:
What happens if the `root_dir` overlaps? Should we break or we handle fusing them ?

Reply:
Good call out. We should definitely error out in that case. I've addressed it here :)